ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -552,6 +553,9 @@ public long numberOfPersistentQueries() {
     return this.livePersistentQueries.size();
   }
 
+  public boolean hasCapacityForPersistentQueries(final long numQueries, final long queryLimit) {
```

Comment:
I would move this to a utility function somewhere in ksql-rest-app. Feels out of place in the engine.